pkg/workflows/wasm/host/module.go

```diff
@@ -185,6 +186,44 @@ func NewModule(modCfg *ModuleConfig, binary []byte) (*Module, error) {
 		return nil, err
 	}
 
+	err = linker.FuncWrap(
+		"env",
+		"log",
+		func(caller *wasmtime.Caller, ptr int32, ptrlen int32) {
+			b, innerErr := safeMem(caller, ptr, ptrlen)
+			if innerErr != nil {
+				logger.Errorf("error calling log: %s", err)
+				return
+			}
+
+			entry, fields, innerErr := wasm.NewLogCodec().Decode(b)
```

Comment:
It would be ideal if we could use the zap json encoder instead of something custom, but if this is truly necessary then we can make it more efficient. There is no need construct a codec each time since decoding is a pure func.

Reply:
I don't see an alternative to this: we tried to use the existing zap json encoder but this turned out to be surprisingly hard to do: the basic problem is that while you can encode the entries just fine, you won't be able to decode it on the other end to pass it to the underlying logger.

Reply:
I'm not understanding the issue. Zap outputs standard JSON with well known fields. What is different about what you are doing?

Or more importantly, do we even need to decode it at all? Could we just pipe it through serialized instead?

Reply:
>Could we just pipe it through serialized instead

This was the initial approach that we tried, but as far as I can see from the API that's not possible.

The lowest level API that I could find for zap was the zapcore.Core interface: https://pkg.go.dev/go.uber.org/zap/zapcore#Core

This requires us to pass in an entry + some fields.

I guess we could pack the serialized bytes into an entry if that's what you mean?

Reply:
I think I might be able to use the zapcore jsonEncoder and just adjust the decoder to its format. Will try this